docs/adrs/003-Retry_endpoint_design.md

```diff
@@ -32,58 +31,75 @@ The endpoint takes a job or list of job ids and then attempts to resubmit them t
+* Blocking and single submit to HTCondor. It should be fine as it returns relatively quickly
   
 ## Retry_jobs behavior
-* Submitting multiple jobs uses the `run_job` endpoint, and is blocking (NOT OK!)
+* Submitting multiple jobs for retry serially calls the same code path
+used for running a single job and blocks until all jobs have been
+submitted to the condor queue. This can cause issues if the
+network drops, and makes the narrative not aware of the state of
+the retry. Submitting 100 jobs currently takes 9 seconds, and that
+is a lot of time for things to go wrong. 
+* (Follow up: Hopefully the making the narrative aware of the state of the retry will be mitigated by the narrative backend. It just blocks on the call anyway, with the default service timeout, which I think is something wacky like half an hour. As long as the user doesn't kill the kernel at that time, all should be well. Of course, if it were me, and it looked frozen for more than a couple minutes, I'd probably restart. )
+* Multiple in-flight retries are allowed.
 
 ### Desired Behavior
-* Prevent multiple in-flight retries to prevent the user from wasting their own resources (and the queues resources)
-* Add retry_count to retried jobs as well to aid in more book-keeping in a new field called `retry_number`
-* Non blocking job submission for submitting multiple jobs, possibly via using `run_job_batch` (requires refactor of run_job_batch)
-* One single submission to HTCondor instead of multiple job submissions
-* Ability to gracefully handle jobs with children
-* Ability to handle database consistentcy during retry failure
+* Create a retry_jobs field, and a toggle to have a list of retried jobs and to ensure DB consistency ( Ability to handle database consistency during retry failure. (see thread https://github.com/kbase/execution_engine2/pull/383#discussion_r638341940))
+* Add failure conditions in the `run()` method and see if any of those should be checked prior to starting jobs
+* Prevent retry if resolving `retry_parent` results in two or more of the same job_ids in a retry_jobs request (see thread https://github.com/kbase/execution_engine2/pull/383#discussion_r640907736)  ( Or silently just run the one retry job and report the results to both positions in the list )
 
+### Can be done with RunJob changes
+* * Non blocking job submission for submitting multiple jobs, possibly via using `run_job_batch` (requires refactor of run_job_batch)
+* * One single submission to HTCondor instead of multiple job submissions
 
-### Questions
+#### Won't do
+* Prevent multiple in-flight retries of the same original job to prevent the user from wasting their own resources (and the queues resources)
+* Add retry_number field
 
-#To Be Answered
+## New priority
+* Create a retry_jobs field, and expose list in api, and a T/F completeness toggle
+* Non blocking job submission / (Possibly htcondor submit)
+* Add failure conditions in run method
```

Comment:
More specifically, look for checks that happen in the run() method and see if they're handled sanely, esp. if we thread starting jobs.

E.g.
- look for quick checks that could be moved prior to creating database records, and for which we should fail the entire submission.
- for checks where it doesn't make sense or is infeasible to perform prior to creating database records and potentially starting a thread to complete processing asynchronously, ensure that the database records are updated correctly if they fail. This applies to exception handling as well.